Middleware/InfoMiddleware.cs

```diff
@@ -0,0 +1,126 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;using Microsoft.Extensions.Logging;
+using Theorem.ChatServices;
+using Theorem.Models;
+using Theorem.Providers;
+
+namespace Theorem.Middleware
+{
+    public class InfoMiddleware : IMiddleware, ISummonable
+    {
+        /// <summary>
+        /// Pattern used to match messages.
+        /// </summary>
+        public string MentionRegex
+        { 
+            get 
+            {
+                return @"(ordered|disabled)?";
+            } 
+        }
+
+        /// <summary>
+        /// verb used to summon this middleware.
+        /// </summary>
+        public static string SummonVerb
+        { 
+            get 
+            {
+                return "info";
+            } 
+        }
+
+        public string Usage
+        {
+            get
+            {
+                return @"info (ordered|disabled)
+  - ordered: print enabled middleware in execution order
+  - disabled: print all disabled middleware
+  - *empty*: print all middleware in alphabetical order, (d) = disabled";
+            }
+        }
+
+        /// <summary>
+        /// Alphabetically ordered list of running middlewares to print.
+        /// </summary>
+        private readonly String _alphabeticallyOrderedMiddleware;
+
+        /// <summary>
+        /// List of running middlewares ordered by execution order to print.
+        /// </summary>
+        private readonly String _orderedMiddleware;
+
+        /// <summary>
+        /// List of disabled middlewares to print.
+        /// </summary>
+        private readonly String _disabledMiddleware;
+
+        private IEnumerable<IChatServiceConnection> chatServiceConnections;
+
+        public InfoMiddleware(ILogger<InfoMiddleware> logger,
+            ConfigurationSection configuration,
+            IEnumerable<IChatServiceConnection> chatServiceConnections,
+            BotMetadataProvider botMetadataProvider)
+        {
+            var orderedMiddlewareList = botMetadataProvider.RunningMiddlewares.Where(mw => mw.Enabled && mw.ExecutionOrderNumber != 0);
+            var unorderedMiddlewareList = botMetadataProvider.RunningMiddlewares.Where(mw => mw.Enabled && mw.ExecutionOrderNumber == 0 && mw.Configured);
```

Comment:
I'm trying to keep lines under 100 chars in length

Reply:
got it, in that case you might wanna consider looking into whether you can commit vscode workspace settings that add a ruler for everyone checking out the code. I believe you should be able to do so.

Reply:
I don't like committing IDE/workflow-specific configs, but I can add a guideline to the readme.